Automattic/jetpack
# Admin: Replace user locale with WordPress.com user locale

Fixes https://github.com/Automattic/wp-calypso/issues/50312

#### Changes proposed in this Pull Request:

* Gets the connected user's WP.com locale preference from the connected user data XMLRPC request.
* Installs the selected locale if needed.
* Stores the selected locale as a user option (so that it's the locale returned by `get_user_locale()`).
* Hides the profile language dropdown and replaces it with a link.

#### Jetpack product discussion


https://github.com/Automattic/wp-calypso/issues/50312

#### Does this pull request change what data or activity we track or use?


Nope.

#### Testing instructions:






- Open wp-admin on an Atomic site. If AMP is activated, make sure you're not using the "Standard" template (see https://github.com/Automattic/jetpack/pull/19828).
- On /wp-admin/options-general.php, change your site language.
- Note that wp-admin changes to the language you chose.
- On /wp-admin/profile.php, note that the language dropdown is set to "Site Default".

<img width="603" alt="Screen Shot 2021-04-13 at 2 55 43 PM" src="https://user-images.githubusercontent.com/1689238/114605728-6336ce00-9c68-11eb-8e5f-3dd77d41dd7a.png">

- Checkout this branch using Jetpack Beta.
- Clear your cache using this wpcom/v2 DELETE endpoint in [the API console](https://developer.wordpress.com/docs/api/console/): /sites/{your blog id}/transients/jetpack_connected_user_data_{your user id}
- Note that wp-admin changes to the account language selected at https://wordpress.com/me/account
- In Users > Profile, note that the language dropdown has been replaced with a link to WP.com:

<img width="544" alt="Screen Shot 2021-04-14 at 6 02 20 PM" src="https://user-images.githubusercontent.com/1689238/114786085-92723b80-9d4b-11eb-87ab-9e90474c85ce.png">

- Try changing your WP.com account language to a language that you haven't used on your site before, and make sure it loads correctly.
- Repeat these steps on a non-Atomic Jetpack site and verify that there is no change.
- Try changing your WP.com account language to a language that isn't supported in wp-admin, like Norsk. Wp-admin should default to US English.

Known issue: non-admin users can't install new languages. See: https://github.com/Automattic/jetpack/issues/19829
